templates/default/InstallChocolatey.ps1.erb

```diff
@@ -47,6 +47,16 @@ $tempDir = Join-Path $chocTempDir "chocInstall"
 if (![System.IO.Directory]::Exists($tempDir)) {[System.IO.Directory]::CreateDirectory($tempDir)}
 $file = Join-Path $tempDir "chocolatey.zip"
 
+try {
+  # This should work in .NET 4 where .NET 4.5 is installed as an inplace upgrade
+  # Enable TLS1.2 (3072)
+  [System.Net.ServicePointManager]::SecurityProtocol |= 3072
+  # Enable TLS1.1 (768)
+  [System.Net.ServicePointManager]::SecurityProtocol |= 768
+} catch {
+  Write-Warning "Unable to set PowerShell to use TLS 1.2 and TLS 1.1 due to old .NET Framework installed. Please upgrade to at least .NET Framework 4.5 and PowerShell v3 for this to work appropriately."
+}
```

Comment:
I'm sorry you spent so much time researching this. https://github.com/puppetlabs/puppetlabs-chocolatey/blob/master/templates/InstallChocolatey.ps1.erb#L63-L76